### PR TITLE
Improve type safety with `noImplicitAny`

### PR DESCRIPTION
**Pull request by jerelmiller**
Partially addresses #1047

Enables the `noImplicitAny` TS rule and fixes all the TS errors as a result. In addition to the fixes, I provided a few improvements:

- Added codegen for the local schema to improve type safety of client-only queries
- Added [`ts-reset`](https://www.totaltypescript.com/ts-reset) to fix some edge case weirdness
- Some minor refactoring on `Queries` and `Mutations` views that avoids the need for 2 queries and instead uses a plain `array.find(...)` to get the selected query.

### CODE REVIEW

File: src/extension/tab/helpers.ts [hunk 1]
```diff
@@ -29,30 +28,39 @@ export function getQueries(
 
       queries.push({
         document,
-        source: document?.loc?.source,
```
Comment (jerelmiller) [inline src/extension/tab/helpers.ts:32]: This is not used when [getting the query string](https://github.com/apollographql/apollo-client-devtools/pull/1052/files#diff-e846862c9de29c06c715196483bc3b28ed25a6741f2b846549112dd97128ff0eR133).

File: src/extension/tab/helpers.ts [hunk 2]
```diff
@@ -64,7 +72,6 @@ export function getMutations(mutationsObj): QueryInfo[] {
     return {
       document: mutation,
       variables,
-      source: mutation?.loc?.source,
```
Comment (jerelmiller) [inline src/extension/tab/helpers.ts:67]: Previously this `source` was used to print the `mutationString`, but was inconsistent with the way queries were printed, which just used the `print` function from `graphql` on the `document`. For consistency, this was updated to do the same.

File: src/extension/tab/hook.ts [hunk 3]
```diff
@@ -148,34 +155,37 @@ function initializeHook() {
...
-    } = JSON.parse(payload);
+    } = JSON.parse(payload ?? "") as {
+      operation: string;
+      operationName: string | undefined;
+      variables: JSONObject | undefined;
+      fetchPolicy: FetchPolicy;
+    };
 
     const queryAst = gql(query);
+    const clonedQueryAst = JSON.parse(JSON.stringify(queryAst)) as DocumentNode;
```
Comment (phryneas) [inline src/extension/tab/hook.ts:173]:
```suggestion
    const clonedQueryAst = JSON.parse(JSON.stringify(queryAst)) as DocumentNode & { definitions: OperationDefinitionNode[]};

```

File: src/extension/tab/hook.ts [hunk 4]
```diff
@@ -148,34 +155,37 @@ function initializeHook() {
...
 
-        return acumm;
-      },
-      []
-    );
+      return acumm;
+    }, []);
 
-    clonedQueryAst.definitions = filteredDefinitions;
+    (clonedQueryAst as any).definitions = filteredDefinitions;
```
Comment (phryneas) [inline src/extension/tab/hook.ts:189]:
```suggestion

    clonedQueryAst.definitions = filteredDefinitions;

```

File: src/Relay.ts [hunk 5]
```diff
@@ -29,12 +30,8 @@ class Relay extends EventTarget {
...
-    return new CustomEvent(message, { detail: {} });
-  };
-
-  public broadcast = (message: MessageObj) => {
-    let event = this.createEvent(message.message);
+  public broadcast = <TPayload = any>(message: MessageObj<TPayload>) => {
+    let event = new CustomEvent(message.message, { detail: message });
 
     if (message?.to) {
       let destination = message.to;
```
Comment (phryneas) [inline src/Relay.ts:37]: I think the next line after this (`event.detail["to"] = destination;`) can go away?

**Review by phryneas — COMMENTED** [review]
Some small nitpicks, but also a big THANK YOU!

File: src/application/components/Explorer/postMessageAuthHelpers.ts [hunk 6]
```diff
@@ -60,7 +60,7 @@ export const handleAuthenticationPostMessage = ({
       "apolloStudioEmbeddedExplorerEncodedApiKey"
```
[the remark this answers is not in the document]
Comment (jerelmiller) [reply src/application/components/Explorer/postMessageAuthHelpers.ts:60]: I noticed that too. I'll go back through in a separate PR and do a few more refactorings like this 🙂

File: src/application/components/Explorer/postMessageAuthHelpers.ts [hunk 7]
```diff
@@ -12,7 +12,7 @@ export const hasGraphRefBeenAuthenticated = (graphRef: string): boolean => {
     "apolloStudioEmbeddedExplorerEncodedApiKey"
   );
   const partialEmbedApiKeys = partialEmbedApiKeysString
-    ? JSON.parse(partialEmbedApiKeysString)
+    ? (JSON.parse(partialEmbedApiKeysString) as Record<string, unknown>)
```
[the remark this answers is not in the document]
Comment (jerelmiller) [reply src/application/components/Explorer/postMessageAuthHelpers.ts:15]: Good catch!

File: src/extension/tab/hook.ts (hunk 3 above)
Comment (jerelmiller) [reply src/extension/tab/hook.ts:173]: Good catch!

File: src/extension/tab/hook.ts (hunk 4 above)
Comment (jerelmiller) [reply src/extension/tab/hook.ts]: [769db5e](https://github.com/apollographql/apollo-client-devtools/pull/1052/commits/769db5ec81199098ebfcac49146af056bd4effc5)

File: src/extension/tab/hook.ts (hunk 3 above)
Comment (jerelmiller) [reply src/extension/tab/hook.ts]: [769db5e](https://github.com/apollographql/apollo-client-devtools/pull/1052/commits/769db5ec81199098ebfcac49146af056bd4effc5)

File: src/Relay.ts (hunk 5 above)
Comment (jerelmiller) [reply src/Relay.ts:37]: Good catch!

Comment (jerelmiller) [reply src/Relay.ts:37]: [6482da7](https://github.com/apollographql/apollo-client-devtools/pull/1052/commits/6482da7c9923e794e0b9765fcb60f13387868d91)

**Review by phryneas — APPROVED** [review]

**Pull request merged**